### ISSUE TITLE
Separate database schema for sensitive/personal data?

### ISSUE THREAD

**Issue by philosophicles** [camdram/camdram#441]
Came up via an email discussion by @hoyes in February, which I just rediscovered. 

The suggestion was to isolate the small portion of our database that contains personal user data (stuff we don't expose via APIs etc) in a separate schema or DB that has different access credentials. 

Most of our database is exposed via APIs anyway and if somebody found a way to dump those tables via some injection attack or whatever, we wouldn't be that bothered. Email addresses/names and perhaps a few other things would matter though, so perhaps should be secured differently.

**Comment by CHTJonas** [comment]
I guess this can't be a bad thing to do but I wonder how necessary it actually is? If we're using a modern up-to-date framework with up-to-date JS/CSS/PHP/whatever dependencies is there enough of a risk factor to warrant the complication of a separate DB/username/password? Happy if people think this is actually a good idea in which case I'll shut up and be quiet (!) but I'm hesitant that the cost/benefit of this justifies doing it.

Historically has Camdram been prone to SQL injection, XSS and such?

**Comment by GKFX** [comment]
Would be worth considerig what actually counts as private; socs such as Porter’s Log, who are moving into film, are annonymous and so even the list of authorized users on a soc/show becomes private information. (Memebridge is another example.)
If we were sufficiently rigorous most of the DB’s tables would move into the “private” side, which would kind of defeat the point.

**Comment by hoyes** [comment]
I can't find the original e-mail thread about this now so not sure what my original thought process was... I'm inclined to agree with you both now.

I referred to something like this recently in #400 as a possible way of providing more realistic test data for non-admin developers (by means of a cleansed DB dump). But yea it's probably more complicated than that to resolve any foreign keys than span sensitive/non-sensitive data, and probably isn't worth the kerfuffle anyway.

Closing... can always reopen if any of us remembers a good reason to do this...
